### PR TITLE
Firefly-886: add user agent based on tag to firefly network calls

### PR DESCRIPTION
**Pull request by robyww**
### Firefly-886: add user agent based on tag to firefly network calls
 
 - gets the firefly tag 
 - show it in the version dialog
 - use the tag to make a user agent for net work calls

_ticket:_ https://jira.ipac.caltech.edu/browse/FIREFLY-886

#### Testing

- add a tag to the branch such as `release-2033.44.33` (that you will delete later)
- build the branch
- You should be the tag in the about dialog and any network call will log the user agent as `firefly/2033.44.33`

### CODE REVIEW

**Comment by gpdf** [comment]
A clarifying question: if an application (like SOFIA, or the Rubin Portal a/k/a "suit") is built on top of Firefly, will it just be the Firefly tag that's reported?

**Comment by robyww** [comment]
> A clarifying question: if an application (like SOFIA, or the Rubin Portal a/k/a "suit") is built on top of Firefly, will it just be the Firefly tag that's reported?

It will be `firefly/2033.44.33 (suit/1.2.3)`

File: src/firefly/java/edu/caltech/ipac/util/download/URLDownload.java [hunk 1]
```diff
@@ -45,6 +49,30 @@
...
+        String tag = v.getBuildGitTagFirefly();
+        if (StringUtils.isEmpty(tag)) return unknown;
+        Matcher m = Pattern.compile("[1-9]").matcher(tag);
+        if (!m.find()) return unknown;
+        int firstIdx= m.start();
+        int lastIdx= m.start();
+        while (m.find()) lastIdx= m.start();
+        try {
+            return "Firefly/" + tag.substring(firstIdx,lastIdx+1);
+        } catch (Exception e) {
```
Comment (loitly) [inline src/firefly/java/edu/caltech/ipac/util/download/URLDownload.java:71]: This logic may leads to wrong value.  It ignores the trailing 0s which is significant.  Try something like this:
`jshell> Pattern.compile("\\d+(\\.\\d+)+").matcher("release-2021.13.30.dev").results().findFirst().get().group();`

**Review by loitly — APPROVED** [review]
Changes look fine.  Look at my comment about pattern.  If you disagree, it's fine, too.  Tag is internally controlled.  Not a big deal.

File: src/firefly/java/edu/caltech/ipac/util/download/URLDownload.java (hunk 1 above)
Comment (robyww) [reply src/firefly/java/edu/caltech/ipac/util/download/URLDownload.java:71]: I did forget the `0` but you are so much better at regular expressions. I am sure this is much better but I will play with it to understand.

**Comment by gpdf** [comment]
"It will be firefly/2033.44.33 (suit/1.2.3)" 

+1, I like that.

**Pull request merged**